clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/producer/ProducerSendWhileDeletionTest.java

```diff
@@ -157,6 +164,42 @@ public void testSendWithRecreatedTopic() throws Exception {
         }
     }
 
+    @Timeout(90)
+    @ClusterTest
+    public void testSendWhileTopicGetRecreated() {
+        int maxNumRecreatTopicAttempts = 10;
+        List<Uuid> topicIds = new CopyOnWriteArrayList<>();
+        var recreateTopicFuture = CompletableFuture.runAsync(() -> {
+            for (int i = 1; i <= maxNumRecreatTopicAttempts; i++) {
+                Uuid topicId = recreateTopic();
+                if (topicId != Uuid.ZERO_UUID) {
```

Comment:
Should we compare using `equals()`:

```suggestion
                if (!topicId.equals(Uuid.ZERO_UUID)) {
```

Reply:
i updated the test a bit and don't need this anymore